src/main/java/br/com/sbk/sbking/core/Board.java

```diff
@@ -76,4 +76,11 @@ public boolean equals(Object obj) {
         return true;
     }
 
+    public void givebackCardToHand(Map<Card, Direction> cardDirectionMap) {
```

Comment:
I think it is giveBack... with an uppercase B. (EDIT: in fact, it is giveBackCard**s**ToHand**s**)
Also, you probably only need to sort after the `for`. There will be another `for` in `Direction.values`, but that number of elements will be constant. The current one increases with the size of the map. (EDIT: The `for` you wrote is constant too (max 2*trick_size) cards. Ignore what I said.)
I know, I just told you to not optimize these things, but this one is really easy, hahaha :laughing: